llama_index/response_synthesizers/google/generativeai/base.py

```diff
@@ -0,0 +1,94 @@
+import logging
+from typing import Any, List, Sequence, cast
+
+from llama_index.bridge.pydantic import BaseModel  # type: ignore
+from llama_index.prompts.mixin import PromptDictType
+from llama_index.response_synthesizers.base import BaseSynthesizer
+from llama_index.types import RESPONSE_TEXT_TYPE
+from llama_index.vector_stores.google.generativeai import google_service_context
+
+_logger = logging.getLogger(__name__)
+_import_err_msg = "`google.generativeai` package not found, please run `pip install google-generativeai`"
+_separator = "\n\n"
+
+
+class SynthesizedResponse(BaseModel):
+    answer: str
+    attributed_passages: List[str]
+    answerable_probability: float
+
+
+class GoogleTextSynthesizer(BaseSynthesizer):
+    _client: Any
+    _answer_style: int
+
+    def __init__(self, answer_style: int = 1, **kwargs: Any):
+        try:
+            import google.ai.generativelanguage as genai
+
+            import llama_index.vector_stores.google.generativeai.genai_extension as genaix
+        except ImportError:
+            raise ImportError(_import_err_msg)
+
+        super().__init__(
+            service_context=google_service_context,
+            output_cls=SynthesizedResponse,
+            **kwargs,
+        )
+        self._client = genaix.build_text_service()
```

Comment:
just to confirm, the genaix should take care of OAuth for the user?

Reply:
Yes, because genaix calls genai, which is the low-level library.